lib/chef/knife/windows_cert_generate.rb

```diff
@@ -75,7 +76,7 @@ def prompt_for_passphrase
       end
 
       def generate_certificate rsa_key
-        @hostname = "*"
+        @hostname = Socket.gethostname
```

Comment:
Minor nit: on Windows, you don't need reverse dns to get the fqdn, for domain-joined systems under the covers you just call [GetComputerNameEx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724301(v=vs.85).aspx) with name format `ComputerNameDnsFullyQualified`. Or call DsCrackNames with a cached name value obtained from WMI or other sources. In one case, it's an LDAP call to the DC (I think) that is cached so you actually don't have to go off the machine and the other also involves no network interaction.